container_opts_unix.go

```diff
@@ -61,30 +75,39 @@ func withRemappedSnapshotBase(id string, i Image, uid, gid uint32, readonly bool
 		if err != nil {
 			return err
 		}
+		if remapper == remapperSnapshotter {
+			opts = append(opts,
+				snapshots.WithLabels(map[string]string{
+					"containerd.io/snapshot/uidmapping": fmt.Sprintf("%d:%d:%d", 0, uid, 65535),
+					"containerd.io/snapshot/gidmapping": fmt.Sprintf("%d:%d:%d", 0, gid, 65535),
+				}))
+		}
 		if _, err := snapshotter.Stat(ctx, usernsID); err == nil {
-			if _, err := snapshotter.Prepare(ctx, id, usernsID); err == nil {
+			if _, err := snapshotter.Prepare(ctx, id, usernsID, opts...); err == nil {
 				c.SnapshotKey = id
 				c.Image = i.Name()
 				return nil
 			} else if !errdefs.IsNotFound(err) {
 				return err
 			}
 		}
-		mounts, err := snapshotter.Prepare(ctx, usernsID+"-remap", parent)
+		mounts, err := snapshotter.Prepare(ctx, usernsID+"-remap", parent, opts...)
```

Comment:
This doesn't seem necessary if the snapshotter is supporting the mapping. I think a better approach would be to leave the remapping functions alone and just provide a helper which can set the labels and just use `WithNewSnapshot`

Reply:
Okay. I'm thinking of an implementation that uses a switch statement in `WithRemappedSnapshot()` and `WithRemappedSnapshotView()` to call `WithNewSnapshot()` based on the `remapper`. Would this be a better approach or should I leave all of the remapping functions alone?

Reply:
ping @dmcgowan